src/domain/templates/components/TemplateSelectors/SubspaceTemplateSelector.tsx

```diff
@@ -1,26 +1,55 @@
 import { Box, Button, Chip, Skeleton } from '@mui/material';
-import { useField } from 'formik';
+import { useField, useFormikContext } from 'formik';
 import React, { FC, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { BlockSectionTitle, Caption } from '@/core/ui/typography';
 import ImportTemplatesDialog from '../Dialogs/ImportTemplateDialog/ImportTemplatesDialog';
 import SystemUpdateAltIcon from '@mui/icons-material/SystemUpdateAlt';
 import { LibraryIcon } from '@/domain/templates/LibraryIcon';
-import { SpaceLevel, TemplateDefaultType, TemplateType } from '@/core/apollo/generated/graphql-schema';
-import { useSpaceDefaultTemplatesQuery, useTemplateNameQuery } from '@/core/apollo/generated/apollo-hooks';
+import { SpaceLevel, TemplateDefaultType, TemplateType, VisualType } from '@/core/apollo/generated/graphql-schema';
+import {
+  useSpaceDefaultTemplatesQuery,
+  useTemplateNameQuery,
+  useTemplateContentLazyQuery,
+} from '@/core/apollo/generated/apollo-hooks';
 import { Identifiable } from '@/core/utils/Identifiable';
 import Gutters, { GuttersProps } from '@/core/ui/grid/Gutters';
 import useUrlResolver from '@/main/routing/urlResolver/useUrlResolver';
+import { getVisualByType } from '@/domain/common/visual/utils/visuals.utils';
+import { VisualModel } from '@/domain/common/visual/model/VisualModel';
+import ConfirmationDialog from '@/core/ui/dialogs/ConfirmationDialog';
+
+export type BasicVisualUrlModel = {
+  avatar?: string;
+  cardBanner?: string;
+};
 
 interface SubspaceTemplateSelectorProps extends GuttersProps {
   name: string;
+  onTemplateVisualsLoaded?: (visualUrls: BasicVisualUrlModel) => void;
 }
 
-export const SubspaceTemplateSelector: FC<SubspaceTemplateSelectorProps> = ({ name, ...rest }) => {
+const getVisualUrls = (visuals: VisualModel[]): BasicVisualUrlModel => {
+  const avatar: string | undefined = getVisualByType(VisualType.Avatar, visuals)?.uri;
```

Comment:
It's not exactly the same result because your way is not even setting the property if it's `undefined`, but it doesn't really matter if it's undefined, I think something like this would be simpler and easier to read: 
```
return {
   avatar: getVisualByType(...),
   cardBanner: getVisualByType(...),
};
```